src/Sentry/GlobalSessionManager.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Linq;
+using System.Net.NetworkInformation;
+using Sentry.Extensibility;
+using Sentry.Internal;
+
+namespace Sentry
+{
+    // AKA client mode
+    internal class GlobalSessionManager : ISessionManager
+    {
+        private readonly SentryOptions _options;
+
+        public Session? CurrentSession { get; private set; }
+
+        public GlobalSessionManager(SentryOptions options)
+        {
+            _options = options;
+        }
+
+        public Session? StartSession()
+        {
+            if (CurrentSession is not null)
+            {
+                _options.DiagnosticLogger?.LogWarning(
+                    "Starting a new session while an existing one is still active."
+                );
+
+                // End previous session (TODO: should this be abnormal instead?)
+                EndSession(SessionEndStatus.Exited);
+            }
+
+            var release = ReleaseLocator.Resolve(_options);
+            if (string.IsNullOrWhiteSpace(release))
+            {
+                // Release health without release is just health (useless)
+                _options.DiagnosticLogger?.LogError(
+                    "Failed to start a session because there is no release information."
+                );
+
+                return null;
+            }
+
+            var environment = EnvironmentLocator.Resolve(_options);
+
+            // TODO: proper distinct id
```

Comment:
This is how cocoa/java does, the 'installation id': https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryInstallation.m

just a guid we write once